diff --git a/auto_nodetitle.module b/auto_nodetitle.module
index 30e6e4f..af7fd05 100644
--- a/auto_nodetitle.module
+++ b/auto_nodetitle.module
@@ -47,6 +47,9 @@ function auto_nodetitle_form_node_form_alter(&$form, &$form_state, $form_id) {
 function auto_nodetitle_node_submit($node, $form, &$form_state) {
   $setting = auto_nodetitle_get_setting($node->type);
   if ($setting == AUTO_NODETITLE_ENABLED || ($setting == AUTO_NODETITLE_OPTIONAL && empty($form_state['values']['title']))) {
+    // Allow fields to load additional data not available in form data. This is
+    // how node_preview() is doing it too.
+    _field_invoke_multiple('load', 'node', array($node->nid => $node));
     auto_nodetitle_set_title($node);
   }
 }
@@ -57,6 +60,9 @@ function auto_nodetitle_node_submit($node, $form, &$form_state) {
 function auto_nodetitle_node_presave($node) {
   // If not yet done, generate the title now.
   if (auto_nodetitle_is_needed($node)) {
+    // Allow fields to load additional data not available in form data. This is
+    // how node_preview() is doing it too.
+    _field_invoke_multiple('load', 'node', array($node->nid => $node));
     auto_nodetitle_set_title($node);
   }
 }
